addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2052,6 +2052,9 @@ export class OdooEditor extends EventTarget {
         if (!sel.anchorNode) {
             show = false;
         }
+        const linkNode = getInSelection(this.document, 'a');
+        const unlinkButton = this.toolbar.querySelector('#unlink');
+        unlinkButton && unlinkButton.classList.toggle('d-none', !linkNode);
```

Comment:
You are right that the problem is that those lines are not executed in the case of mobile. However, I do not think this is the right fix.

I suggest changing the lines below instead:
```diff
-            if (show !== undefined && !this.isMobile) {
-                this.toolbar.style.visibility = show ? 'visible' : 'hidden';
-            }
-            if (show === false) {
-                return;
+            if (!this.isMobile) {
+                if (show !== undefined) {
+                    this.toolbar.style.visibility = show ? 'visible' : 'hidden';
+                }
+                if (show === false) {
+                    return;
+                }
```
Basically, wrap those lines inside a `!this.isMobile` condition so that `show` is basically ignored in the case where `this.isMobile` is `true`. This matches the functional expecations because the toolbar is always displayed in mobile.

Reply:
@dmo-odoo Applied all the suggested changes. Please look into it.

Reply:
Thanks !